components/docker/component.go

```diff
@@ -20,33 +21,46 @@ const (
 	defaultTimeout = 300
 )
 
+type ManagerOutput struct {
+	components.JSONImporter
+
+	Host remoteComp.HostOutput `json:"host"`
+}
+
 type Manager struct {
+	pulumi.ResourceState
+	components.Component
+
 	namer namer.Namer
-	host  *remoteComp.Host
+	opts  []pulumi.ResourceOption
 
-	opts []pulumi.ResourceOption
+	Host *remoteComp.Host `pulumi:"host"`
 }
 
-func NewManager(e config.CommonEnvironment, host *remoteComp.Host, opts ...pulumi.ResourceOption) (*Manager, pulumi.Resource, error) {
-	manager := &Manager{
-		namer: e.CommonNamer.WithPrefix("docker"),
-		host:  host,
-		opts:  opts,
-	}
+func NewManager(e config.CommonEnvironment, host *remoteComp.Host, opts ...pulumi.ResourceOption) (*Manager, error) {
+	return components.NewComponent(e, host.Name()+"-dockerManager", func(comp *Manager) error {
+		comp.namer = e.CommonNamer.WithPrefix("docker")
+		comp.Host = host
+		comp.opts = opts
```

Comment:
`pulumi.Parent(comp)` needs to be added here if you want to be able to `DependOn` the component itself later.

Reply:
Not sure here, should I add `pulumi.Parent(comp)` to `comp.opts`? Wouldn't that mean that `comp` is the parent of `comp`?